india_compliance/gst_india/client_scripts/e_waybill_actions.js

```diff
@@ -33,13 +33,41 @@ function setup_e_waybill_actions(doctype) {
             if (frm.doc.__onload?.e_waybill_info?.is_generated_in_sandbox_mode)
                 frm.get_field("ewaybill").set_description("Generated in Sandbox Mode");
 
-            if (
-                frm.doc.docstatus != 1 ||
-                frm.is_dirty() ||
-                frm.doc.e_waybill_status === "Not Applicable" ||
-                !is_e_waybill_applicable(frm)
-            )
+            if (!is_e_waybill_api_enabled(frm) || frm.is_dirty()) return;
+
+            if (frm.doc.docstatus == 0) {
+                frm.add_custom_button(
+                    __("Applicability Status"),
+                    () =>
+                        show_e_waybill_applicability_status(
+                            frm,
+                            is_e_waybill_applicable(frm, true)
+                        ),
+                    "e-Waybill"
+                );
                 return;
+            }
+
+            if (frm.doc.docstatus == 1 && !is_e_waybill_applicable(frm, true)) {
```

Comment:
- Wrapper function for show message
- Extend this to cover generatable messages
- refactor and better variable names
- Use Generatable API